orca-core/src/main/java/com/netflix/spinnaker/orca/TaskResolver.java

```diff
@@ -113,7 +113,8 @@ public Task getTask(@Nonnull Class<? extends Task> taskType) {
   @Nonnull
   public Class<? extends Task> getTaskClass(@Nonnull String taskTypeIdentifier) {
     try {
-      return getTask(taskTypeIdentifier).getClass();
+      Task task = getTask(taskTypeIdentifier);
+      return (Class<? extends Task>) task.getExtensionClass();
```

Comment:
When the extension is proxied, `TaskExecution.type` class was being serialized as a `Proxy` object.  This fixes that, and allows `TaskTypeDeserializer` to deserialize the proxied object to the correct Task type.